core/osm_connector.py

```diff
@@ -372,3 +396,170 @@ def _query_stops(self):
+
+        """
+        shape = []
+
+        ways = []
+        for member in route_variant.members:
+            if isinstance(member, overpy.RelationWay):
+                if not member.role == "platform":
+                    ways.append(member.ref)
+
+        shape_sorter = []
+        node_geography = {}
+
+        for way in ways:
+            # Obtain geography (nodes) from original query result set
+            nodes = query_result_set.get_ways(way).pop().get_nodes()
+
+            # Prepare data for sorting and geographic information of nodes
+            way_nodes = []
+            for node in nodes:
+                way_nodes.append(node.id)
+                node_geography[node.id] = {'lat': float(
+                    node.lat), 'lon': float(node.lon)}
+
+            if len(shape_sorter) == 0:
+                shape_sorter.extend(way_nodes)
+            elif shape_sorter[-1] == way_nodes[0]:
+                del shape_sorter[-1]
+                shape_sorter.extend(way_nodes)
+            elif shape_sorter[-1] == way_nodes[-1]:
+                del shape_sorter[-1]
+                shape_sorter.extend(reversed(way_nodes))
+            elif shape_sorter[0] == way_nodes[0]:
+                del shape_sorter[0]
+                shape_sorter.reverse()
+                shape_sorter.extend(way_nodes)
+            elif shape_sorter[0] == way_nodes[-1]:
+                del shape_sorter[0]
+                shape_sorter.reverse()
+                shape_sorter.extend(reversed(way_nodes))
+            else:
+                sys.stderr.write(
+                    "Route has non-matching ways: " + str(self) + "\n")
+                sys.stderr.write(
+                    "  Problem at: http://osm.org/way/" + str(way) + "\n")
+                break
+
+        for sorted_node in shape_sorter:
+            shape.append(node_geography[sorted_node])
+
+        return shape
+
+    def _is_valid_stop_candidate(self, stop):
+        """Helper function to check if a stop candidate has a valid tagging
+
+        Returns True or False
+
+        """
+        if 'public_transport' in stop.tags:
+            if stop.tags['public_transport'] == 'platform':
+                return True
+            elif stop.tags['public_transport'] == 'station':
+                return True
+        elif 'highway' in stop.tags:
+            if stop.tags['highway'] == 'bus_stop':
+                return True
+        elif 'amenity' in stop.tags:
+            if stop.tags['amenity'] == 'bus_station':
+                return True
+        return False
+
+    def _get_names_for_unnamed_stops(self):
+
+        """Intelligently guess stop names for unnamed stops by sourrounding
+        street names and amenities.
+
+        Caches stops with newly guessed names.
+
+        """
+        # Loop through all stops
+        for stop in self.stops.values():
+
+            # If there is no name, query one intelligently from OSM
+            if stop.name == "[" + self.stop_no_name + "]":
```

Comment:
I see this is how you identify the stops that haven't got a name. Maybe it is better use `None` here and only change all that are still none `None` after this method ran (or didn't ran) to the variable from the config file?